src/cpp/include/openvino/genai/whisper_generation_config.hpp

```diff
@@ -0,0 +1,82 @@
+// Copyright (C) 2023-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "openvino/genai/tokenizer.hpp"
+#include "openvino/runtime/compiled_model.hpp"
+
+namespace ov {
+namespace genai {
+
+/**
+ * @brief Structure to keep generation config parameters.
+ *
+ * Generic parameters:
+ * @param max_length the maximum length the generated tokens can have. Corresponds to the length of the input prompt +
+ *        `max_new_tokens`. Its effect is overridden by `max_new_tokens`, if also set.
+ * @param max_new_tokens the maximum numbers of tokens to generate, excluding the number of tokens in the prompt.
+ * max_new_tokens has priority over max_length.
+ * @param ignore_eos if set to true, then generation will not stop even if <eos> token is met.
+ * @param eos_token_id token_id of <eos> (end of sentence)
+ */
+class OPENVINO_GENAI_EXPORTS WhisperGenerationConfig {
```

Comment:
should this generation config be derived from LLM's generation config? 
As I remember, Whisper (and other ASRs) are commonly used with beam search as Kate said.

Reply:
@slyalin is solving the same problem with LoRA. What's the difference between `WhisperGenerationConfig` and `GenerationConfig`?

Reply:
Agreed to inherit from `GenerationConfig` when sampling enabled for `WhisperPipeline`